lib/pool.js

```diff
@@ -215,39 +215,71 @@ class Core extends PoolFunctions {
       }
     }
     // Start query execution..
-    scope.connection.query(scope.sqlCommand, (err, rows) => {
-      // Register query
-      this.poolActivity[this.currentSessionId()]++;
-      // Check for any errors
-      if (err) {
-        // Assign sql commands
-        err.sql = scope.sqlCommand;
-        // Release connection
-        scope.connection.release();
-        // Callback [ERROR]
-        if (typeof scope.returnCallback === 'function') {
-          scope.returnCallback([], err);
-          return;
+    if(typeof scope.values !== undefined){
+      scope.connection.query(scope.sqlCommand, scope.values, (err, rows) => {
+        // Register query
+        this.poolActivity[this.currentSessionId()]++;
+        // Check for any errors
+        if (err) {
+          // Assign sql commands
+          err.sql = scope.sqlCommand;
+          // Release connection
+          scope.connection.release();
+          // Callback [ERROR]
+          if (typeof scope.returnCallback === 'function') {
+            scope.returnCallback([], err);
+            return;
+          } else {
+            return;
+          }
         } else {
-          return;
+          // Release connection
+          scope.connection.release();
+          // Callback [SUCCESS]
+          if (typeof scope.returnCallback === 'function') {
+            scope.returnCallback(rows, {});
```

Comment:
What's the signature of `returnCallback`? Everywhere else the 2nd param appears to be an Error object or string.

Reply:
Apologies for the late response, what do you mean by signature? The `returnCallback` first parameter should always be an array while the second will be an object (containing error information)